test/truefi2/lines-of-credit/DebtToken.test.ts

```diff
@@ -0,0 +1,166 @@
+import { expect, use } from 'chai'
+import { MockProvider, solidity } from 'ethereum-waffle'
+import { BigNumberish, Wallet } from 'ethers'
+
+import { beforeEachWithFixture, parseEth, setupTruefi2 } from 'utils'
+
+import { DebtToken, DebtToken__factory, MockTrueCurrency, MockTrueCurrency__factory } from 'contracts'
+
+use(solidity)
+
+describe('DebtToken', () => {
+  enum DebtTokenStatus {
+    Defaulted = 4,
+    Liquidated
+  }
+
+  const payback = async (wallet: Wallet, amount: BigNumberish) => token.mint(debtToken.address, amount)
+  const debt = parseEth(1)
+
+  let lender: Wallet
+  let borrower: Wallet
+  let safu: Wallet
+  let debtToken: DebtToken
+  let token: MockTrueCurrency
+  let poolAddress: string
+  let provider: MockProvider
+
+  beforeEachWithFixture(async (wallets, _provider) => {
+    [lender, borrower, safu] = wallets
+    provider = _provider
+
+    const { standardPool } = await setupTruefi2(lender, provider)
+    poolAddress = standardPool.address
+    token = MockTrueCurrency__factory.connect(await standardPool.token(), lender)
+    debtToken = await new DebtToken__factory(lender).deploy()
+    await debtToken.initialize(standardPool.address, lender.address, borrower.address, safu.address, debt)
+  })
+
+  describe('Constructor', () => {
+    it('correctly takes token from pool', async () => {
+      expect(await debtToken.token()).to.equal(token.address)
+    })
+
+    it('sets pool address', async () => {
+      expect(await debtToken.pool()).to.equal(poolAddress)
+    })
+
+    it('sets borrowers debt', async () => {
+      expect(await debtToken.debt()).to.equal(debt)
+    })
+
+    it('sets erc20 params', async () => {
+      expect(await debtToken.name()).to.equal('TrueFi Debt Token')
+      expect(await debtToken.symbol()).to.equal('DEBT')
+      expect(await debtToken.decimals()).to.equal(18)
+    })
+
+    it('mints tokens for the lender', async () => {
+      expect(await debtToken.balanceOf(lender.address)).to.equal(debt)
+      expect(await debtToken.totalSupply()).to.equal(debt)
+    })
+
+    it('sets status to Defaulted', async () => {
+      expect(await debtToken.status()).to.equal(DebtTokenStatus.Defaulted)
+    })
+  })
+
+  describe('liquidate', () => {
+    it('reverts if liquidated twice', async () => {
+      await debtToken.connect(safu).liquidate()
+      await expect(debtToken.connect(safu).liquidate())
+        .to.be.revertedWith('DebtToken: Current status should be Defaulted')
+    })
+
+    it('reverts if not called by liquidator', async () => {
+      await expect(debtToken.liquidate())
+        .to.be.revertedWith('DebtToken: Caller is not the liquidator')
+    })
+
+    it('sets status to liquidated', async () => {
+      await debtToken.connect(safu).liquidate()
+      expect(await debtToken.status()).to.equal(DebtTokenStatus.Liquidated)
+    })
+  })
+
+  describe('Redeem', () => {
+    beforeEach(async () => {
+      await token.mint(borrower.address, parseEth(100))
+    })
```

Comment:
Same as above: We should add back tests to make sure this isn't called if Status < 4, if possible without too much hassle.

Reply:
I'll go ahead and resolve this comment too since we'd have to test against an unreachable state.

However, following this comment: https://github.com/trusttoken/smart-contracts/pull/942/files#r703661740, the contract should still guard against the unreachable state with a require check to make sure it remains blocked, even if we accidentally introduced new code to violate our invariant.